src/distilabel/tasks/text_generation/self_instruct.py

```diff
@@ -149,11 +156,17 @@ def to_argilla_dataset(self, dataset_row: Dict[str, Any]) -> "FeedbackDataset":
         )  # type: ignore
         # Then we just return the `FeedbackDataset` with the fields, questions, and metadata properties
         # defined above.
-        return rg.FeedbackDataset(
+        dataset = rg.FeedbackDataset(
             fields=fields,
             questions=questions,  # type: ignore
             metadata_properties=metadata_properties,  # Note that these are always optional
         )
+        dataset: (
+            FeedbackRecord | List[FeedbackRecord] | FeedbackDataset
```

Comment:
Does this `Union` format works for python 3.8? I'm not sure, but if that's the case would love to update it all around the codebase

Reply:
good catch